clar2wasm/src/standard/standard.wat

```diff
@@ -1727,6 +1728,69 @@
         )
     )
 
+    (func $buff-to-uint-be (param $offset i32) (param $length i32) (result i64 i64)
+        (local $mask_lo i64) (local $mask_hi i64) (local $double v128)
+        (if (i32.gt_u (local.get $length) (i32.const 16))
+            (then (call $runtime-error (i32.const 5)))
+        )
+
+        (if (i32.eqz (local.get $length))
+            (then (return (i64.const 0) (i64.const 0)))
+        )
+
+        ;; SAFETY: this function works because we already have data in the memory,
+        ;;         otherwise we could have a negative offset.
+        (local.set $offset (i32.sub (i32.add (local.get $offset) (local.get $length)) (i32.const 16)))
+
+        ;; we compute masks for the low and high part of the resulting integer
+        (local.set $mask_lo
+            (select
+                (i64.const -1)
+                (local.tee $mask_hi
+                    (i64.shr_u (i64.const -1) (i64.extend_i32_u (i32.and (i32.mul (local.get $length) (i32.const 56)) (i32.const 56))))
+                )
+                (i32.ge_u (local.get $length) (i32.const 8))
+            )
+        )
+        (local.set $mask_hi (select (local.get $mask_hi) (i64.const 0) (i32.gt_u (local.get $length) (i32.const 8))))
+
+        ;; we load both low and high part at once, and rearrange the bytes for endianness
+        (local.set $double
+            (i8x16.swizzle
+                (v128.load (local.get $offset))
```

Comment:
Does the vector load have any alignment requirements? Like will it cause an error if the source address is not aligned?

Reply:
Vector load is exactly the same as `i32.load` or `i64.load`: it reads 128 bits at once, except that it does not rearrange the bytes order.